### PR TITLE
Promote emptyconfigmap key e2e test to conformance test

### PR DESCRIPTION
**Pull request by globervinodhn**
**What type of PR is this?** /kind cleanup
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
> /kind feature
> /kind flake

**What this PR does / why we need it**: To promote EmptyConfigMapKey e2e test to conformance test

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**: Convert e2e test to conformance test

**Does this PR introduce a user-facing change?**: NONE

```release-note
NONE
```
cc @spiffxp 

/area conformance

### CODE REVIEW

**Comment by oomichi** [comment]
This changes an e2e test which verifies failure case behavior to a conformance test.
Is it worth verifying failure case behavior as conformance tests?

**Comment by globervinodhn** [comment]
> This changes an e2e test which verifies failure case behavior to a conformance test.
> Is it worth verifying failure case behavior as conformance tests?

The test in question tries to verify if a pod is created when the configmap has an empty key. The pod creation should not happen as per the specification.

**Comment by mgdevstack** [comment]
IMO as this test is just verifying the obvious reason for **ConfigMap creation failure** and has nothing to do with Volume(misleading testName), it may not make sense to promote it to Conformance instead keep it as e2e only. 

Please correct me on this @oomichi -
An e2e test should be promoted to Conformance if it verifies the PodSpec/k8s-component fundamental behavior differently, verifying various available Spec-options etc ultimately leading to an increase in API coverage.

Following observed points from this PR.
1. Test name renaming like "should fail to create empty-key configMap" based on test behavior.
2. Do not promote it to conformance.

**Comment by oomichi** [comment]
@mgdevstack

Yeah, I have the same thinkig with you and thank you for describing it clearly.
Maybe it is better to get opinions from more people who are interested in conformance tests.

**Comment by spiffxp** [comment]
We're interested in _prioritizing_ pod functionality, but this reads to me like behavior I would expect of any kubernetes cluster

File: test/e2e/common/configmap.go [hunk 1]
```diff
@@ -124,7 +124,12 @@ var _ = Describe("[sig-node] ConfigMap", func() {
 		})
 	})
 
-	It("should fail to create configMap in volume due to empty configmap key", func() {
+	/*
+	   Release : v1.14
+	   Testname: ConfigMap with empty key, in volume
+	   Description: Attempt to create a configmap in volume. The creation should fail due to empty configmap key.
+	*/
+	framework.ConformanceIt("should fail to create configMap in volume due to empty configmap key", func() {
```
Comment (spiffxp) [inline test/e2e/common/configmap.go]: This is not volume-specific, can you drop that from the testname, description, and It?

**Comment by spiffxp** [comment]
/approve
/lgtm
/assign @bgrant0607 @smarterclayton 
for conformance approval

**Comment by spiffxp** [comment]
The pull-kubernetes-e2e-kops-aws failure is no longer relevant as the job has been moved to non-blocking

**Comment by smarterclayton** [comment]
/approve

ConfigMap validation is a key part of conformance.

**Pull request merged**